controller/space_iterations_blackbox_test.go

```diff
@@ -0,0 +1,558 @@
+			ci.Data.Attributes.UserActive = ptr.Bool(true)
+			fxt := tf.NewTestFixture(t, s.DB, tf.CreateWorkItemEnvironment())
+			svc := testsupport.ServiceAsUser("Iteration-Service", *fxt.Identities[0])
+			ctrl := NewSpaceIterationsController(svc, s.db, s.Configuration)
+			// when
+			resp, iter := test.CreateSpaceIterationsCreated(t, svc.Context, svc, ctrl, fxt.Spaces[0].ID, ci)
+			// then
+			compareWithGoldenAgnostic(t, filepath.Join(s.testDir, "create", "ok_with_force_active.payload.res.golden.json"), iter)
+			compareWithGoldenAgnostic(t, filepath.Join(s.testDir, "create", "ok_with_force_active.headers.res.golden.json"), resp)
+		})
+
+		t.Run("with optional values", func(t *testing.T) {
+			// given
+			fxt := tf.NewTestFixture(t, s.DB, tf.CreateWorkItemEnvironment())
+			iterationName := "Sprint #22"
+			iterationDesc := "testing description"
+			ci := newCreateSpaceIterationPayload(iterationName, &iterationDesc)
+			svc, ctrl := s.SecuredController(*fxt.Identities[0])
+			// when
+			_, c := test.CreateSpaceIterationsCreated(t, svc.Context, svc, ctrl, fxt.Spaces[0].ID, ci)
+			// then
+			assert.NotNil(t, c.Data.ID)
+			assert.NotNil(t, c.Data.Relationships.Space)
+			assert.Equal(t, fxt.Spaces[0].ID.String(), *c.Data.Relationships.Space.Data.ID)
+			assert.Equal(t, *c.Data.Attributes.Name, iterationName)
+			assert.Equal(t, *c.Data.Attributes.Description, iterationDesc)
+
+			// create another Iteration with nil description
+			iterationName2 := "Sprint #23"
+			ci = newCreateSpaceIterationPayload(iterationName2, nil)
+			_, c = test.CreateSpaceIterationsCreated(t, svc.Context, svc, ctrl, fxt.Spaces[0].ID, ci)
+			assert.Equal(t, *c.Data.Attributes.Name, iterationName2)
+			assert.Nil(t, c.Data.Attributes.Description)
+		})
+	})
+
+	s.T().Run("failure", func(t *testing.T) {
+		t.Run("missing space", func(t *testing.T) {
+			// given
+			ci := newCreateSpaceIterationPayload("Sprint #21", nil)
+			svc, ctrl := s.SecuredController()
+			// when/then
+			test.CreateSpaceIterationsNotFound(s.T(), svc.Context, svc, ctrl, uuid.NewV4(), ci)
+		})
+
+		t.Run("duplicate iteration", func(t *testing.T) {
+			fxt := tf.NewTestFixture(t, s.DB,
+				tf.Identities(1),
+				tf.Iterations(2, func(fxt *tf.TestFixture, idx int) error {
+					if idx == 1 {
+						fxt.Iterations[idx].MakeChildOf(*fxt.Iterations[0])
+					}
+					return nil
+				}))
+			iterationName := fxt.Iterations[1].Name
+			iterationDesc := "duplicate iteration description"
+			ci := newCreateSpaceIterationPayload(iterationName, &iterationDesc)
+			svc, ctrl := s.SecuredController(*fxt.Identities[0])
+			// when
+			test.CreateSpaceIterationsConflict(t, svc.Context, svc, ctrl, fxt.Spaces[0].ID, ci)
+		})
+
+		t.Run("not authorized", func(t *testing.T) {
+			// given
+			ci := newCreateSpaceIterationPayload("Sprint #21", nil)
+			svc, ctrl := s.UnSecuredController()
+			// when/then
+			test.CreateSpaceIterationsUnauthorized(s.T(), svc.Context, svc, ctrl, uuid.NewV4(), ci)
```

Comment:
It would be nice to create a golden file for the error here as well:

```go
res, jerrs := test.CreateSpaceIterationsUnauthorized(s.T(), svc.Context, svc, ctrl, uuid.NewV4(), ci)
compareWithGoldenAgnostic(t, filepath.Join(s.testDir, "create", "unauthorized.payload.res.golden.json"), jerrs)
compareWithGoldenAgnostic(t, filepath.Join(s.testDir, "create", "unauthorized.headers.res.golden.json"), res.Header())
```